rest_framework_json_api/relations.py

```diff
@@ -142,19 +148,48 @@ def to_representation(self, value):
         else:
             pk = value.pk
 
-        # check to see if this resource has a different resource_name when
-        # included and use that name
-        resource_type = None
-        root = getattr(self.parent, 'parent', self.parent)
         field_name = self.field_name if self.field_name else self.parent.field_name
-        if getattr(root, 'included_serializers', None) is not None:
-            includes = get_included_serializers(root)
-            if field_name in includes.keys():
-                resource_type = get_resource_type_from_serializer(includes[field_name])
 
-        resource_type = resource_type if resource_type else get_resource_type_from_instance(value)
+        resource_type = self.get_resource_type_from_included_serializer(field_name)
+        if resource_type is None:
+            resource_type = get_resource_type_from_instance(value)
+
         return OrderedDict([('type', resource_type), ('id', str(pk))])
 
+    def get_resource_type_from_included_serializer(self, field_name):
+        """
+        Given a field_name, check if the serializer has a
+        corresponding included_serializer with a Meta.resource_name property
+
+        Returns the resource name or None
+        """
+        root = getattr(self.parent, 'parent', self.parent) or self.parent
+        root = self.get_root_serializer()
+
+        if root is not None:
+            # accept both singular and plural versions of field_name
+            field_names = [
+                inflection.singularize(field_name),
+                inflection.pluralize(field_name)
+            ]
+            includes = get_included_serializers(root)
+            for field in field_names:
+                if field in includes.keys():
+                    return get_resource_type_from_serializer(includes[field])
+
+        return None
+
+    def get_root_serializer(self):
+        if hasattr(self.parent, 'parent') and self.is_serializer(self.parent.parent):
+            return self.parent.parent
+        elif self.is_serializer(self.parent):
+            return self.parent
+
+        return None
+
+    def is_serializer(self, candidate):
+        return hasattr(candidate, 'included_serializers')
```

Comment:
I had to tell apart the user's serializer and DRF's `ListSerializer` - which are both serializers.. I guess the easiest and least error-prone way is to check for what were looking for anyway, which is `included_serializers`.